docs/tutorial.rst

```diff
@@ -0,0 +1,1842 @@
+.. currentmodule:: reframe.core.pipeline.RegressionTest
+
+================
+ReFrame Tutorial
+================
+
+This tutorial will cover the basic concepts of ReFrame and will get you started with the framework.
+For more specific topics, you should refer to ":doc:`howto`" as well as to the ":doc:`topics`" for an in-depth understanding of some of the framework's concepts.
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 3
+
+Requirements
+============
+
+To run this tutorial you need ``docker`` for the local examples and ``docker compose`` for the examples emulating a Slurm cluster.
+
+The tutorial container images provide already the latest ReFrame version installed.
+For installing a stand-alone version of ReFrame, please refer to the ":doc:`started`" guide.
+
+All tutorial examples are located under the ``reframe-examples`` directory inside the container's working directory.
+
+
+Running the local examples
+--------------------------
+
+To run the local examples, launch the single-node tutorial container by binding mounting the examples:
+
+.. code-block:: bash
+
+   docker run -h myhost -it --mount type=bind,source=(pwd)/examples/,target=/home/user/reframe-examples reframe-tut-singlenode:latest /bin/bash
+
+
+.. _multi-node-setup:
+
+Running the multi-node examples
+-------------------------------
+
+To run the multi-node examples you need first to launch a Slurm pseudo cluster using the provided Docker compose file:
+
+.. code-block:: bash
+
+   docker compose --project-directory=$(pwd) -f examples/tutorial/dockerfiles/slurm-cluster/docker-compose.yml up --abort-on-container-exit --exit-code-from frontend
+
+Once the Docker compose stack is up, you execute the following from a difference console window in order to "log in" in the frontend container:
+
+.. code-block::
+
+   docker exec -it $(docker ps -f name=frontend --format=json | jq -r .ID) /bin/bash
```

Comment:
It may be because I'm running Docker 20.10.16, but this doesn't work as expected. The `--format=json` seems to be problematic, maybe? If I manually grab the `CONTAINER ID` instead of the $() command substitution, it works.

Reply:
I'm on Docker 24.0.6 so that could be the reason; I'll add a note.